### PR TITLE
fix: Datepicker emitEvent error handling on parsing of icelandic language

### PR DESCRIPTION
**Pull request by harshith-venkatesh-freshworks**
## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

File: packages/crayons-core/src/components/datepicker/datepicker.tsx [hunk 1]
```diff
@@ -999,9 +999,41 @@ export class Datepicker {
 
   processValueChange(val, emitChange = false) {
     // show error if not ISO format and not display format
-    const parsedDate = parse(val, this.displayFormat, new Date(), {
-      locale: this.langModule,
-    });
+    let parsedDate;
+    if (this.langModule?.code === 'is' && this.dateFormat === 'dd MMM yyyy') {
```
Comment (rihansiddhi) [inline packages/crayons-core/src/components/datepicker/datepicker.tsx]: @harshith-venkatesh-freshworks Can you move this to a common method and the constants separately as we are duplicating the code here?

File: packages/crayons-core/src/components/datepicker/datepicker.tsx [hunk 2]
```diff
@@ -81,6 +81,52 @@ const getWeekDays = (lang): any => {
...
+const parse = (value, displayFormat, date, langModule) => {
+  if (!value) return value;
+  if (langModule?.locale?.code === 'is' && displayFormat === 'dd MMM yyyy') {
+    return parseIcelandicDate(value, langModule);
+  }
+  return parseDate(value, displayFormat, date, langModule);
+};
+
+const isMatch = (value, displayFormat, langModule) => {
+  return (
```
Comment (arvindanta) [inline packages/crayons-core/src/components/datepicker/datepicker.tsx]: if langmodule is `is`, this will return false. is this the expectation?

Comment (harshith-venkatesh-freshworks) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx]: Yes @arvindanta , as isMatch is not supported completely for icelandic language dates.
Though, we are parsing through mapping, the isMatch method would still throw error for the same

File: packages/crayons-core/src/components/datepicker/datepicker.tsx [hunk 3]
```diff
@@ -918,10 +937,10 @@ export class Datepicker {
     if (
       !isValidFromDate ||
       !isValidToDate ||
-      !isMatch(fromDate, this.displayFormat, {
+      isNoMatch(fromDate, this.displayFormat, {
```
Comment (arvindanta) [inline packages/crayons-core/src/components/datepicker/datepicker.tsx]: can u retain isMatch here. and modify the function above ?.

**Review by arvindanta — APPROVED** [review]

**Review by arvindanta — APPROVED** [review]

**Review by rihansiddhi — APPROVED** [review]

**Pull request merged**